server/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -549,26 +577,61 @@ public Clusters(StreamInput in) throws IOException {
             } else {
                 this.clusterInfo = Collections.emptyMap();
             }
+            if (in.getTransportVersion().onOrAfter(TransportVersions.SEARCH_RESPONSE_CLUSTERS_COUNTERS_ADDED)) {
```

Comment:
Is it possible to avoid serializing the new counters entirely? If yes, then we don't need a new TransportVersion bump.

For local-only searches, they will always be zero and for CCS the count for running, partial, etc are always counted from the underlying SearchResponse.Cluster objects dynamically.

Reply:
Yes, it's possible! Done!